src/AutoMapper/ConfigurationStore.cs

```diff
@@ -364,6 +364,17 @@ public TypeMap FindTypeMapFor(TypePair typePair)
             return typeMap;
         }
 
+        private bool CoveredByObjectMap(TypePair typePair)
+        {
+            return GetMappers().FirstOrDefault(m => m.IsMatch(new ResolutionContext(null, null, typePair.SourceType, typePair.DestinationType, new MappingOperationOptions(), new MappingEngine(this)))) != null;
+        }

```

Comment:
I don't see any other way then to check all the object mappers to determined if it's covered by any of them.  You can't rely on certain ones by explicitly using them, because they could be removed from mappers and give type map not found exceptions when trying to do the mappings after the fact.  Also down the road there might be more added, and they need to be able to be checked as well.  My Automapper.Collections dll adds 2 IObjectMappers into the master list, and it needs to know about them for when I call Persist, and need to map `Mapper.Map<TDTO,Expression<T,bool>>`.  If it's not all the mapping or is some hidden list, or filter rules, I'm going to have to modify my object mappers I added in in order to make them work with conventions.

If you are concerned about performance, well this should happen only once per TypePair.  Once it's found it will be cahced and this will never have to be called again.  And it's isMatch vs Map which takes less time to compute cause it's usually like 1-5 boolean checks.